services/httpd/handler.go

```diff
@@ -834,6 +855,52 @@ func (h *Handler) serveExpvar(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintln(w, "\n}")
 }
 
+// serveTrackRequests will track requests for a period of time.
+func (h *Handler) serveTrackRequests(w http.ResponseWriter, r *http.Request) {
+	var d time.Duration
+	if s := r.URL.Query().Get("seconds"); s == "" {
+		h.httpError(w, "missing required seconds parameter", http.StatusBadRequest)
+		return
+	} else if seconds, err := strconv.ParseInt(s, 10, 64); err != nil {
+		h.httpError(w, err.Error(), http.StatusBadRequest)
+		return
+	} else {
+		d = time.Duration(seconds) * time.Second
+	}
+
+	var closing <-chan bool
+	if notifier, ok := w.(http.CloseNotifier); ok {
+		closing = notifier.CloseNotify()
+	}
+
+	profile := h.requestTracker.TrackRequests()
+
+	timer := time.NewTimer(d)
+	select {
+	case <-timer.C:
+		profile.Stop()
+	case <-closing:
+		// Connection was closed early.
+		profile.Stop()
+		timer.Stop()
+		return
+	}
+
+	w.Header().Add("Connection", "close")
```

Comment:
Anything that uses `CloseNotify()` needs to have this since the HTTP connection can't be reused. I can remove the `CloseNotify()` part, but then a lost connection will result in requests continuing to be tracked after the connection is lost.